server/website/website/db/base/target_objective.py

```diff
@@ -108,9 +134,33 @@ def register(self):
                         self._metric_metadatas[dbms_id] = [(mname, BaseMetric(mname)) for mname
                                                            in sorted(numeric_metrics)]
 
+            LOG.info('Registering user defined target objectives...')
+            dbmss = models.DBMSCatalog.objects.all()
+            for dbms in dbmss:
+                dbms_id = int(dbms.pk)
+                if dbms_id not in self._registry:
+                    self._registry[dbms_id] = {}
+                self._registry[dbms_id][USER_DEFINED_TARGET] = UserDefinedTargetObjective()
```

Comment:
Is it possible to register it on use? For example, in register_udm()? The issue now is that when there is no user defined target, there is still an item in the dict.

Reply:
When there is no user defined target, there is still an item in the dict, but it will not show in the metric list. 
However, the user defined metric option will always show when creating a new session.

Reply:
Will it show in the return value of get_all()?
Referenced here: https://github.com/cmu-db/ottertune/search?q=target_objectives.get_all&unscoped_q=target_objectives.get_all

Reply:
I think so. The User Defined Metric will show as a target objective option when creating a session